metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -71,6 +71,7 @@ class Compilers(
     trees: Trees,
     mtagsResolver: MtagsResolver,
     sourceMapper: SourceMapper,
+    workspaceId: String,
```

Comment:
Why do we need workspaceId? Is this for logging?

Reply:
We don't really need it. We can use URI everywhere. It's attached to things like commands that we produce, so we know where to reroute. And workspace uri is sometimes empty and this cannot be even for tests.